sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -1035,7 +1034,11 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
             m.appendTail(${termResult});
             ${ev.primitive} = ${classNameUTF8String}.fromString(${termResult}.toString());
             ${ev.isNull} = false;
+          } else {
+            ${ev.isNull} = true;
```

Comment:
I found `UnaryExpression` and `BinaryExpression` both use `${eval.isNull}` as the initial value of `${ev.isNull}`, therefore I did the same here.
If it's not a convention I would just use `true` as initial value and remove these two `else { ${ev.isNull} = true;` here.
And do the same thing in `RegExpExtract` codegen as well